File: apps/mobile/src/types/review.ts
export interface Review {
  id: number;
  reviewerId: string;
  targetId: string;
  content: string;
  score: number;
  createdAt: string;
}

export interface CreateReviewRequest {
  targetId: string;
  content: string;
  score: number;
}

export interface ReviewSummary {
  total: number;
  averageScore: number;
  data: Review[];
}

File: apps/api/app/schemas/review.py
from datetime import datetime

from pydantic import BaseModel, Field


class ReviewCreate(BaseModel):
    targetId: str
    content: str
    score: int = Field(..., ge=1, le=5)


class ReviewResponse(BaseModel):
    id: int
    reviewerId: str
    targetId: str
    content: str
    score: int
    createdAt: datetime

    class Config:
        from_attributes = True


class ReviewListResponse(BaseModel):
    total: int
    averageScore: float
    data: list[ReviewResponse]
